R/germline.R

```diff
@@ -61,30 +60,14 @@ repGermline <- function(.data, species = "HomoSapiens") {
   }
 }
 
-germline_single_df <- function(data, species) {
-  # add first allele of V and J genes
-  data["V.first.allele"] <- data %>%
-    select(V.name) %>%
-    apply(MARGIN = 1, FUN = take_first_allele)
-  data["J.first.allele"] <- data %>%
-    select(J.name) %>%
-    apply(MARGIN = 1, FUN = take_first_allele)
-
-  # load V genes
-  v_genes <- load_reference_sequences("IGHV", species)
-
-  # add V genes
-  data <- merge(x = v_genes, y = data, by = "V.first.allele", all.y = TRUE)
-
-  # load J genes
-  j_genes <- load_reference_sequences("IGHJ", species)
-
-  # add J genes
-  data <- merge(x = j_genes, y = data, by = "J.first.allele", all.y = TRUE)
-
+germline_single_df <- function(data, species, sample_name = NA) {
   data %>%
-    mutate(Germline.Sequence = purrr::map2(
-      V.sequence, J.sequence, generate_germline_sequence
+    rowwise() %>%
+    mutate(V.first.allele = take_first_allele(V.name)) %>%
+    merge_reference_sequences("IGHV", species, sample_name) %>%
+    rowwise() %>%
+    mutate(Germline.sequence = generate_germline_sequence(
+      Sequence, V.sequence, V.end, CDR3.start, CDR3.end, J.start
     )) %>%
     return()
```

Comment:
Please edit the code according to the style guide
https://style.tidyverse.org/functions.html#return

Reply:
It is best to avoid right-hand side assignment and return